generator/sbpg/targets/c.py

```diff
@@ -304,12 +304,17 @@ def __init__(self, msg, package_specs):
         self.fields = []
         self.is_real_message = msg.is_real_message
         self.sibling_include = []
+        self.packed_size = 0
+        self.is_fixed_size = True
         #print("Creating message %s" % self.name)
         for f in msg.fields:
             new_field = FieldItem(msg, package_specs, f)
-            if new_field.packed_size == 0 and not self.is_real_message:
-                print("Field %s in message %s: variable length arrays can only exist in real messages, not in embedded types" % (new_field.name, msg.identifier))
-                raise "error"
+            self.packed_size = self.packed_size + new_field.packed_size
+            if new_field.packed_size == 0:
+                self.is_fixed_size = False
+                if not self.is_real_message:
+                    print("Field %s in message %s: variable length arrays can only exist in real messages, not in embedded types" % (new_field.name, msg.identifier))
+                    raise "error"
```

Comment:
Just use `raise Exception` and move error message from the print statement to here, e.g. `raise Exception("...")`